### DISCUSSION TITLE
Optimal settings for stable multiroom streaming

### DISCUSSION THREAD

**Discussion by tobox** [sle118/squeezelite-esp32#425] [General]
I was looking for an open-source multi-room streaming system and bought 4 identical ESP32s with UDA1334A. 

Everything works "almost" but there seem to be some synchronization issues. 

I have been playing around with cross-fade settings, changed buffer sizes, maximum bit rates etc. But it is not clear what all those settings do. I.e.:

- -r <minrate>-<maxrate> - should this be "auto detected or should it be set to 44100 or 48000 for best results?
- -Z - I tried to set it to 96000 (as recommended in the docs) in the web gui but it did not show up in the NVM editor
- -b - I started with 500:2000 and tried some other settings, but found no good explanation what each of the values represents. 

My guess would be that a larger output buffer is only necessary for Bluetooth output, but for I2A it should not make a big difference, correct? Should I increase the stream buffer and decrease the output buffer?

**Comment by wizmo2** [comment]
Multi-room synchronization is handled by the Slim/Squeezebox/Logitech/Lyrion(!!!) server.  There are delay and latency settings in the player configurations.  For help on this I recommend you look to the Squeezebox forum ( https://forums.slimdevices.com )

Not looked into the Squeezelite configurations much.  You typically only need to modified these to maintain a stable stream if you are using high bitrate  audio

**Comment by tobox** [comment]
I know that most parts of the multiroom stuff is handled by the server. I.e. one of my "smart" speakers has a 180ms delay even when using 3.5mm input. I was able to fix that delay using the server.

But I often have issues with intermittent pauses in the stream or similar dropouts of audio. Often it happens on all devices synchronously, but sometimes it happens on just one speaker. 

I found the "System Statistics" option an will now continuously collect the log infos to dig deeper into the problem.
